picard/webservice/api_helpers.py

```diff
@@ -60,102 +60,84 @@ def _wrap_xml_metadata(data):
-             request_mimetype=None):
-        path = self.api_path + "/".join(path_list)
-        return self._webservice.post(self.host, self.port, path, data, handler,
-                                     priority=priority, important=important, mblogin=mblogin,
-                                     queryargs=queryargs, parse_response_type=parse_response_type,
-                                     request_mimetype=request_mimetype)
-
-    def put(self, path_list, data, handler, priority=True, important=False,
-            mblogin=True, queryargs=None, request_mimetype=None):
-        path = self.api_path + "/".join(path_list)
-        return self._webservice.put(self.host, self.port, path, data, handler,
-                                    priority=priority, important=important, mblogin=mblogin,
-                                    queryargs=queryargs, request_mimetype=request_mimetype)
-
-    def delete(self, path_list, handler, priority=True, important=False,
-               mblogin=True, queryargs=None):
-        path = self.api_path + "/".join(path_list)
-        return self._webservice.delete(self.host, self.port, path, handler,
-                                       priority=priority, important=important, mblogin=mblogin,
-                                       queryargs=queryargs)
+    def url_from_path(self, path):
+        url = QUrl(self._base_url)
+        url.setPath(self._base_url.path() + path)
+        return url
+
+    def get(self, path, handler, **kwargs):
+        kwargs['url'] = self.url_from_path(path)
+        kwargs['handler'] = handler
+        return self._webservice.get_url(**kwargs)
+
+    def post(self, path, data, handler, **kwargs):
+        kwargs['url'] = self.url_from_path(path)
+        kwargs['handler'] = handler
+        kwargs['data'] = data
+        kwargs['mblogin'] = kwargs.get('mblogin', True)
+        return self._webservice.post_url(**kwargs)
+
+    def put(self, path, data, handler, **kwargs):
+        kwargs['url'] = self.url_from_path(path)
+        kwargs['handler'] = handler
+        kwargs['data'] = data
+        kwargs['priority'] = kwargs.get('priority', True)
+        kwargs['mblogin'] = kwargs.get('mblogin', True)
+        return self._webservice.put_url(**kwargs)
+
+    def delete(self, path, handler, **kwargs):
+        kwargs['url'] = self.url_from_path(path)
+        kwargs['handler'] = handler
+        kwargs['priority'] = kwargs.get('priority', True)
+        kwargs['mblogin'] = kwargs.get('mblogin', True)
+        return self._webservice.delete_url(**kwargs)
 
 
 class MBAPIHelper(APIHelper):
 
     def __init__(self, webservice):
-        super().__init__(None, None, "/ws/2/", webservice)
-
-    @property
-    def host(self):
         config = get_config()
-        return config.setting['server_host']
+        host = config.setting['server_host']
+        port = config.setting['server_port']
```

Comment:
@phw, this change is incorrect I think, because class is instanced once in https://github.com/metabrainz/picard/blob/8ed7d4343511242d6604ed27c2ee2d2a566af02c/picard/tagger.py#L332
But those values can be modified during runtime.

Reply:
Check https://github.com/metabrainz/picard/pull/2225/commits/9a0bedff20d266f6e32b9b2db460775e38456bb6

Reply:
Great. Makes also cleaner code in the base class